packages/discord.js/src/util/Util.js

```diff
@@ -474,39 +474,40 @@ function basename(path, ext) {
   const res = parse(path);
   return ext && res.ext.startsWith(ext) ? res.name : res.base.split('?')[0];
 }
+
 /**
  * The content to have all mentions replaced by the equivalent text.
  * @param {string} str The string to be converted
  * @param {TextBasedChannels} channel The channel the string was sent in
  * @returns {string}
  */
 function cleanContent(str, channel) {
-  str = str
-    .replace(/<@!?[0-9]+>/g, input => {
-      const id = input.replace(/<|!|>|@/g, '');
-      if (channel.type === ChannelType.DM) {
-        const user = channel.client.users.cache.get(id);
-        return user ? `@${user.username}` : input;
-      }
+  return str.replaceAll(/<(@[!&]?|#)(\d{17,19})>/g, (match, type, id) => {
```

Comment:
I think we should keep it like it was before with `[0-9]+`, instead of having to update the pattern every time ids bump a number (even if that would take many years) but I wouldn't call that a blocker, just a thought

Reply:
`[0-9]+` can match invalid mentions too, we can also use `\d{17,}` so we don't need to increment it every time but it won't match ids with length <17

Reply:
There'll be maximum a 20-digit ID, but that's assuming Discord's database uses `u64` (range up to 18,446,744,073,709,551,615, 20 digits) and not `i64` (range up to 9,223,372,036,854,775,807, 19 digits).

Either way, we would need to wait until 2074 (iirc from my old calculations) for us to run out of `i64` IDs and start needing `u64`.

A maximum range of 19 already surpasses the longevity of this version and many others.